<p>
    Every Puppet <code>metadata.json</code> file should define the following keys:
</p>
<ul>
    <li><code>name</code></li>
    <li><code>version</code></li>
    <li><code>author</code></li>
    <li><code>license</code></li>
    <li><code>summary</code></li>
    <li><code>source</code></li>
    <li><code>dependencies</code></li>
</ul>

<h2>Noncompliant Code Example</h2>
Missing <code>author</code> and <code>license</code> keys:
<pre>
{
  "name": "examplecorp-mymodule",
  "version": "0.0.1",
  "summary": "A module for a thing",
  "source": "https://github.com/examplecorp/examplecorp-mymodule",
  "dependencies": [
    { "name": "puppetlabs/stdlib", "version_requirement": ">=3.2.0 <5.0.0" },
    { "name": "puppetlabs/firewall", "version_requirement": ">= 0.0.4" }
  ]
}
</pre>

<h2>Compliant Solution</h2>
<pre>
{
  "name": "examplecorp-mymodule",
  "version": "0.0.1",
  "author": "Pat",
  "license": "Apache-2.0",
  "summary": "A module for a thing",
  "source": "https://github.com/examplecorp/examplecorp-mymodule",
  "dependencies": [
    { "name": "puppetlabs/stdlib", "version_requirement": ">=3.2.0 <5.0.0" },
    { "name": "puppetlabs/firewall", "version_requirement": ">= 0.0.4" }
  ]
}
</pre>

<h2>See</h2>
<ul>
    <li><a href="http://docs.puppetlabs.com/puppet/latest/reference/modules_publishing.html#write-a-metadatajson-file">Puppet
        Labs documentation - Write a metadata.json File</a></li>
</ul>